presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyPlanWithEmptyInput.java

```diff
@@ -197,6 +205,51 @@ else if (isEmptyNode(rewrittenRight)) {
             return node.replaceChildren(ImmutableList.of(rewrittenLeft, rewrittenRight));
         }
 
+        @Override
+        public PlanNode visitSequence(SequenceNode node, RewriteContext<Void> context)
+        {
+            List<PlanNode> cteProducers = node.getCteProducers();
+            List<PlanNode> newSequenceChildrenList = new ArrayList<>();
+            // Visit in the order of execution
+            for (int i = cteProducers.size() - 1; i >= 0; i--) {
+                PlanNode rewrittenProducer = context.rewrite(cteProducers.get(i));
+                if (!isEmptyNode(rewrittenProducer)) {
```

Comment:
The cte producer which converted to empty values node are removed from sequence node, what about the corresponding cte consumer? How do we deal with them?

Reply:
So we first update the session saying that this is not materialized. If you check the **visitConsumer()** method in this optimizer, it uses the session to detect this. https://github.com/prestodb/presto/pull/22196/files/687eab33a5c17772abef6710c9cf08b4322a8ad4#diff-2341227ca888731a72c9e28c05277e64930ce13aee3ca6af0d744c1e6923c035